crates/wasmtime/src/runtime/vm/traphandlers.rs

```diff
@@ -734,11 +772,84 @@ impl CallThreadState {
     }
 }
 
-// A private inner module for managing the TLS state that we require across
-// calls in wasm. The WebAssembly code is called from C++ and then a trap may
-// happen which requires us to read some contextual state to figure out what to
-// do with the trap. This `tls` module is used to persist that information from
-// the caller to the trap site.
+/// A private inner module managing the state of Wasmtime's thread-local storage
+/// (TLS) state.
+///
+/// Wasmtime at this time has a single pointer of TLS. This single pointer of
+/// TLS is the totality of all TLS required by Wasmtime. By keeping this as
```

Comment:
FYI, this will no longer be true once CM async support is merged given that `concurrent.rs` has a couple of TLS variables (and `wasmtime-wit-bindgen` may also generate code which includes zero or more TLS variables).

Reply:
Hm ok yeah that's going to need to be carefully integrated, "just add a TLS var" breaks portability guarantees we have, so it's not as simple as just relaxing this documentation. We'll have to handle that at some point.